### ISSUE TITLE
Transition model should update `most_recent` after destroy

### ISSUE THREAD

**Issue by hale** [gocardless/statesman#179]
If you destroy the latest transition, there is no 'most_recent' transition for the parent model, breaking the current_state method and scopes

```ruby
class Transition < ActiveRecord::Base
  belongs_to :parent
  after_destroy :update_most_recent 

  private

  def update_most_recent
    parent.history.last.update(most_recent: true)
  end
end
```

Would something like the above work?

**Comment by greysteil** [comment]
Sounds like a good idea to me. If you add a PR for that along with a test then I'm :+1:.

**Comment by greysteil** [comment]
Implemented in #180 - thanks for the suggestion @hale
